packages/actor-rdf-metadata-aggregate-total-items/lib/ActorRdfMetadataAggregateTotalItems.ts

```diff
@@ -0,0 +1,69 @@
+import { ActorRdfMetadataAggregate, IActionRdfMetadataAggregate, IActorRdfMetadataAggregateOutput } from '@comunica/bus-rdf-metadata-aggregate';
+import { IActorArgs, IActorTest } from '@comunica/core';
+import {Record} from "immutable";
+
+/**
+ * A comunica Total Items RDF Metadata Aggregate Actor.
+ */
+export class ActorRdfMetadataAggregateTotalItems extends ActorRdfMetadataAggregate {
+  public records: Record<string,any>[] = [];
+
+  public constructor(args: IActorArgs<IActionRdfMetadataAggregate, IActorTest, IActorRdfMetadataAggregateOutput>) {
+    super(args);
+  }
+
+  public async test(action: IActionRdfMetadataAggregate): Promise<IActorTest> {
+    return true;
+  }
+
+  public async run(action: IActionRdfMetadataAggregate): Promise<IActorRdfMetadataAggregateOutput> {
+    const {metadata, subMetadata} = action
+
+    let newTotalItems: number = 0
+    const isEmptyRecord = (r: Record<string, any>) => Object.keys(r).length === 0
+    const hasTotalItems = (r: Record<string, any>) => r.totalItems !== undefined
+    let resultRecord = {}
+
+    if (!subMetadata) {
```

Comment:
Can you double-check if the semantics of this actor are still the same as https://github.com/comunica/comunica/blob/master/packages/actor-rdf-resolve-quad-pattern-federated/lib/FederatedQuadSource.ts#L226-L235 ?

Because I suspect there will be some difference. E.g. because `Number.isFinite` is never checked in here.